test/JDBC/src/main/java/com/sqlsamples/JDBCTempTable.java

```diff
@@ -0,0 +1,339 @@
+        int num_connections = 10;
+        int num_tables = 5000;
+        
+        /* Create a UDT so that we can test non-ENR temp tables */
+        String connectionString = initializeConnectionString();
+        ArrayList<Connection> cxns = new ArrayList<>();
+        Connection c = DriverManager.getConnection(connectionString);
+        cxns.add(c);
+        c.createStatement().execute("CREATE TYPE my_temp_type FROM int");
+
+        ArrayList<Thread> threads = new ArrayList<>();
+
+        /* Create connections */
+        for (int i = 0; i < num_connections; i++) {
+            Connection connection = DriverManager.getConnection(connectionString);
+            cxns.add(connection);
+            Thread t = new Thread(new Worker(connection, i + 1, num_tables / num_connections, bw));
+            threads.add(t);
+            t.start();
+        }
+        for (Thread t : threads)
+        {
+            t.join();
+        }
+        c.createStatement().execute("DROP TYPE my_temp_type");
+
+        for (Connection cxn : cxns) {
+            cxn.close();
+        }
+    }
+
+    private static void test_oid_buffer(BufferedWriter bw, Logger logger) throws Exception {
+        String connectionString = initializeConnectionString();
+        Connection c = DriverManager.getConnection(connectionString);
+        JDBCCrossDialect cx = new JDBCCrossDialect(c);
+
+        Connection psql = cx.getPsqlConnection("-- psql", bw, logger);
+        int num_connections = 2;
+
+        /*
+         * TEST: After disabling GUC, ensure that the OIDs are not equal, meaning we aren't using
+         * the OID buffer.
+         */
+        Statement alter_guc = psql.createStatement();
+        alter_guc.execute("ALTER DATABASE jdbc_testdb SET babelfishpg_tsql.temp_oid_buffer_size = 0");
+
+        if (check_oids_equal_helper(num_connections)) {
+            bw.write("OID check failed! Oids were equal after disabling guc.");
+            bw.newLine();
+        }
+
+        /*
+         * TEST: Ensure that we can create up to (and no more) than the oid buffer size.
+         */
+        alter_guc.execute("ALTER DATABASE jdbc_testdb SET babelfishpg_tsql.temp_oid_buffer_size = 10");
+
+        /* We need a new connection here to pick up the updated guc. */
+        Connection c2 = DriverManager.getConnection(connectionString);
+        Statement s = c2.createStatement();
+
+        try {
+            for (int i = 0; i < 11; i++) {
+                String queryString = "CREATE TABLE #tab" + i + " (a int)";
+                s.execute(queryString);
+            }
+        } catch (Exception e) {
+            if (!e.getMessage().startsWith("Unable to allocate oid for temp table.")) {
+                bw.write(e.getMessage());
+                bw.newLine();
+            }
+        }
+
+        /* If the table was created, throw an error. */
+        ResultSet rs = s.executeQuery("SELECT * FROM babelfish_get_enr_list() WHERE relname = \'#table_cant_be_created\'");
```

Comment:
not clear, why do expect to find this name for failed case?

Reply:
In psql, '#' is not an allowed character in table names.